scenarios/kubernetes_e2e.py

```diff
@@ -309,14 +329,81 @@ def sig_handler(self, _signo, _frame):
         check('docker', 'stop', self.container)
 
 
-def cluster_name(cluster, build):
+def cluster_name(cluster, build, domain=None):
     """Return or select a cluster name."""
     if cluster:
-        return cluster
-    if len(build) < 20:
-        return 'e2e-%s' % build
-    return 'e2e-%s' % hashlib.md5(build).hexdigest()[:10]
+        clu = cluster
+    elif len(build) < 20:
+        clu = 'e2e-%s' % build
+    else:
+        clu = 'e2e-%s' % hashlib.md5(build).hexdigest()[:10]
+    if domain:
+        clu = '%s.%s' % (clu, domain)
+    return clu
+
+def get_aws_zone(zone):
+    if zone:
+        return zone
+    return random.choice([
+        'us-west-1a',
+        'us-west-1c',
+        'us-west-2a',
+        'us-west-2b',
+        'us-east-1a',
+        'us-east-1d',
+        #'us-east-2a',
+        #'us-east-2b',
+    ])
+
+def build_kops(workspace, mode):
```

Comment:
It seems like we should be doing this build inside the docker container -- either by moving this logic into the kops e2e runner or better yet kubetest